### PR TITLE
:sparkles:(lld): rework coin control UI

### PR DESCRIPTION
**Pull request by LucasWerey**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - Rework of the UI of the coin Control under newArch behind ordinals FF

### 📝 Description

Rework of the coin control UI using ordinals assets from simplehash
+ little changes on the modal component to be able to test it with jest
+ ResizeObserver mocked in jest to avoid error with modals

this is the first PR another one will come to add simplehash to the coinControl and display assets. Currently it's not like the Figma because we don't have the correct components but it's close

There is some TS ignore that already exist so I didn't change that because it's more about BTC implem

https://github.com/user-attachments/assets/1c0f0e3b-afad-4181-bb07-b7ef6088518d

image

### ❓ Context

- **JIRA or GitHub link**: [LIVE-12597](https://ledgerhq.atlassian.net/browse/LIVE-12597)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-12597]: https://ledgerhq.atlassian.net/browse/LIVE-12597?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/CoinControlModal/components/Row/NoOrdinalsRow.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { SplitAddress, Cell } from "~/renderer/components/OperationsList/AddressCell";
+import { useTranslation } from "react-i18next";
+
+type Props = {
+  outputIndex: number;
+  hash: string;
+  address: string;
+};
+
+export const NoOrdinalsRow: React.FC<Props> = ({ outputIndex, hash, address }) => {
```
Comment (themooneer) [inline apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/CoinControlModal/components/Row/NoOrdinalsRow.tsx:12]: while this file represent the `NoOrdinalsRow` main component, it's better to export it as a default.

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/CoinControlModal/components/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { BitcoinOutput } from "@ledgerhq/coin-bitcoin/lib/types";
+import { useTranslation } from "react-i18next";
+
+type Props = {
+  onClickLink: () => void;
+  onClose: () => void;
+  returning?: BitcoinOutput; // will be used in the future
+};
+
+export const Footer: React.FC<Props> = ({ onClickLink, onClose }) => {
```
Comment (themooneer) [inline apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/CoinControlModal/components/Footer.tsx:12]: you can export it as default

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/CoinControlModal/components/Row/index.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            <Tooltip
+              placement="top"
+              content={
+                <TooltipContainer>
+                  <Icons.Information size="S" />
+                  <Trans i18nKey={"bitcoin.cannotSelect.pending"} />
+                </TooltipContainer>
+              }
+            ></Tooltip>
+          ) : last ? (
```
Comment (themooneer) [inline apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/CoinControlModal/components/Row/index.tsx:82]: avoid nested ternary for better readability and less cognitive complexity, replace it with a renderer function (you can memoize its ref with a callback if you want) and call it here. 

Here is a suggestion (function name is up to you)

```jsx
const renderSelectionInfo = ( ) => {

if(unconfirmed) return <Tooltip
              placement="top"
              content={
                <TooltipContainer>
                  <Icons.Information size="S" />
                  <Trans i18nKey={"bitcoin.cannotSelect.pending"} />
                </TooltipContainer>
              }
            /> 

if(last) return <Tooltip
              placement="top"
              content={
                <TooltipContainer>
                  <Icons.Information size="S" />
                  <Trans i18nKey={"bitcoin.cannotSelect.last"} />
                </TooltipContainer>
              }
            >
              <Checkbox isChecked disabled />
            </Tooltip>
  
return <Checkbox isChecked={!excluded} onChange={handleClick} />;
}

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/CoinControlModal/components/Row/index.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+          {unconfirmed ? (
+            <Tooltip
+              placement="top"
+              content={
+                <TooltipContainer>
+                  <Icons.Information size="S" />
+                  <Trans i18nKey={"bitcoin.cannotSelect.pending"} />
+                </TooltipContainer>
+              }
+            ></Tooltip>
```
Comment (themooneer) [inline apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/CoinControlModal/components/Row/index.tsx:81]: can <Tooltip ../> be auto closed ? do you release need a closing tag for it ?

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/CoinControlModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+  onClose,
+  onChange,
+  updateTransaction,
+  onClickLink,
+}) => {
+  const { t } = useTranslation();
+  const { colors } = useTheme();
+
+  return (
+    <>
```
Comment (themooneer) [inline apps/ledger-live-desktop/src/newArch/features/Collectibles/Ordinals/components/CoinControlModal/index.tsx:42]: you don't need a fragment here

File: apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCollectiblesAnalytics.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import { track } from "~/renderer/analytics/segment";
+
+export enum AnalyticsPage {
```
Comment (themooneer) [inline apps/ledger-live-desktop/src/newArch/features/Collectibles/hooks/useCollectiblesAnalytics.ts:3]: it's not the responsibility of a hook to export enums, i recommend to extract this to a local `enums.ts` file

File: .changeset/wet-socks-film.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+---
+"@ledgerhq/coin-bitcoin": patch
+"ledger-live-desktop": patch
```
[the remark this answers is not in the document]
Comment (LucasWerey) [reply .changeset/wet-socks-film.md:3]: I don't know what kind of change should be considered as a minor. We usually use pach for that

**Review by themooneer — APPROVED** [review]

**Review by ypolishchuk-ledger — APPROVED** [review]

**Review by hedi-edelbloute — APPROVED** [review]

**Review by Wozacosta — APPROVED** [review]

**Pull request merged**